wasm/types.go

```diff
@@ -70,6 +75,11 @@ func readElemType(r io.Reader) (ElemType, error) {
 	return ElemType(b), err
 }
 
+func writeElemType(w io.Writer, b ElemType) error {
```

Comment:
looking at this code and the code above (`readElemType`) as well as most of the code in there, it seems we could have a couple of interfaces:
```go
type WASMMarshaler interface {
    MarshalWASM(w io.Writer) error
}

type WASMUnmarshaler interface {
    UnmarshalWASM(r io.Reader) error
}
```

(or `WASMWriter` and `WASMReader`? they don't follow typical (un)marshaler interfaces...)

that a bunch of types in here could implement...

not sure this is a good idea nor whether it should be done as part of this PR...